include/pisa/util/string_util.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <algorithm>
+#include <string>
+#include <utility>
+#include <locale>
+
+namespace pisa {
+namespace util {
+std::string ltrim(const std::string &&str) {
```

Comment:
Why is the rvalue ref constant? You either want to move value (non-const) or copy, then use const ref.

I think we can either:
1. Take rvalue and return object.
2. Take const ref and return `string_view`. We can do that because the result will always point to a substring. Then, one can create a string from that view if a copy is desired. Otherwise, one can operate on a view.

There's also an option of implementing both interfaces if that makes sense.